lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java

```diff
@@ -64,31 +82,50 @@ public static long getPID() {
         final Long result = Long.valueOf(new String(returnBytes, StandardCharsets.UTF_8).trim());
         process.destroy();
         return result;
-      } catch (final IOException e) {
+      } catch (final Exception e) {
+        LOG.log(Level.SEVERE, "Unable to determine PID", e);
+        return -1;
+      }
+    } else if (isWindows()) {
+      try {
+        final Process process = new ProcessBuilder()
+            .command("powershell.exe", "-Command", "wmic process where processid=$pid get parentprocessid")
```

Comment:
During testing, I noticed that this fails on my machine because my profile prints out stuff before the printout this looks for. I replaced it 

```
.command("powershell.exe", "-NoProfile", "-Command", "wmic process where processid=$pid get parentprocessid")
```

which fixed the issue. I will merge it this way.